File: jarvis.py
import pyttsx3
import speech_recognition as sr
import pyaudio
import datetime
import os
#import random 
from requests import get
import wikipedia as wp
import webbrowser
from googlesearch import search
import pywhatkit as w

engine = pyttsx3.init('sapi5')
voices = engine.getProperty('voices')
print(voices[0].id)
engine.setProperty('voices',voices[0].id)

#text to speech
def speak(audio):
    engine.say(audio)
    engine.runAndWait()

#To convirt voice to text
def takecommand():
    r = sr.Recognizer()
    with sr.Microphone() as source:
        print("I am listening you...")
        r.pause_threshold = 1
        audio = r.listen(source)

    try:
        print("Recognizing...")
        query =r.recognize_google(audio, language='en-in')
        print(f"user said: {query}")

    except Exception as e:
        speak("Say that again please...")
        return "none"
    return query

#To wish
def wish():
    hour = int(datetime.datetime.now().hour)

    if hour>=0 and hour<=12:
        speak("Good Morning")
    elif hour>12 and hour<18:
        speak("Good Afternoon")
    elif hour>=18 and hour<20:
        speak("Good Evening")
    else:
        speak("Good Night")
    
    speak("Sir I am Jack How can I help you sir")

if __name__ == "__main__":
    #speak("Hi Soubhagya This is advance jarvice How can I help you")
    wish()
    if 1:

        query = takecommand().lower()

        #logic buliding for nodepad

        if "open nodepad" in query:
            # npath = "C:\\WINDOWS\\system32\\nodepad.exe"
            # os.startfile(npath)
            os.system("start notepad")
        
        elif "open powershell" in query:
            # cpath = "C:\\WINDOWS\\system32\\cmd.exe"
            # os.startfile(cpath)
            os.system("start powershell")
        
        elif "fuck you" in query:
            speak("Buro... ki... maa... ki... chud...")

        elif "wikipedia" in query:
            #speak("Searching wikipedia...")
            print("Searching wikipedia...")
            query = takecommand().lower()
            query = query.replace("wikipedia",query)
            #results = wp.search(query,results = 5)
            results=wp.summary(query,sentences=2)
            #speak("wikipedia said that")
            speak(results)
            print(results)
        
        elif "open youtube" in query:
            webbrowser.open("www.youtube.com")
        
        elif "open facebook" in query:
            webbrowser.open("www.facebook.com")

        elif "open instagram" in query:
            webbrowser.open("www.instagram.com")
        
        elif "open google" in query:
            #speak("Searching wikipedia...")
            print("Searching google...")
            que = takecommand().lower()
            #query = query.replace("wikipedia",query)
            #results = webbrowser.open(que)
            results = w.search(que)
            #results=wp.summary(query,sentences=2)
            #speak("wikipedia said that")
            speak(results)
            print(results)

        elif "google" in query:
            que = takecommand().lower()
 
            for j in search(que):
                print(j)

        elif "send message" in query:
            w.sendwhatmsg("+917363928885","hi I am Jack your asistant",17,32)

        elif "play youtube" in query:
            que = takecommand().lower()
            w.playonyt(que)

File: jarvis.c
#include <iostream>
#include <string>
#include <ctime>
#include <cstdlib>
#include <windows.h>
#include <sapi.h>
#include <speechapi_cxx.h>

using namespace std;
using namespace Microsoft::CognitiveServices::Speech;

void speak(const string& audio) {
    ISpVoice* pVoice = NULL;
    CoInitialize(NULL);
    if (FAILED(CoCreateInstance(CLSID_SpVoice, NULL, CLSCTX_ALL, IID_ISpVoice, (void**)&pVoice))) {
        return;
    }
    pVoice->Speak(std::wstring(audio.begin(), audio.end()).c_str(), SPF_IS_XML, NULL);
    pVoice->Release();
    CoUninitialize();
}

string takeCommand() {
    // Placeholder for speech recognition logic
    // This requires a speech recognition library or API
    // For now, we will return a dummy command
    return "open notepad"; // Simulating user input
}

void wish() {
    time_t now = time(0);
    tm* localtm = localtime(&now);
    int hour = localtm->tm_hour;

    if (hour >= 0 && hour < 12) {
        speak("Good Morning");
    } else if (hour >= 12 && hour < 18) {
        speak("Good Afternoon");
    } else {
        speak("Good Night");
    }

    speak("Sir I am Jack How can I help you sir");
}

int main() {
    wish();
    string query = takeCommand();

    if (query.find("open notepad") != string::npos) {
        system("start notepad");
    } else if (query.find("open cmd") != string::npos) {
        system("start cmd");
    }

    return 0;
}

